modules/src/main/java/org/archive/modules/fetcher/FetchHTTPRequest.java

```diff
@@ -168,6 +171,12 @@ public ServerCacheResolver(ServerCache serverCache) {
     protected HttpHost targetHost;
     protected boolean addedCredentials;
     protected HttpHost proxyHost;
+
+    // SOCKS5 proxy configuration
+    protected String socksProxyHost;
+    protected Integer socksProxyPort;
+    protected Boolean useSocksProxy;
```

Comment:
```suggestion
    protected boolean useSocksProxy;
```
The constructor always sets this and the null case doesn't seem meaningful so maybe this could be a primitive?